### PR TITLE
feat(server): enable/disable request feature for specific roles in projects

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
In this PR I added `RequestRoles` slice to `Project` type.
The specified roles in the `RequestRoles` will not be able to publish without making requests.

### CODE REVIEW

File: server/pkg/project/builder.go [hunk 1]
```diff
@@ -84,3 +86,8 @@ func (b *Builder) Publication(publication *Publication) *Builder {
 	b.p.publication = publication
 	return b
 }
+
+func (b *Builder) RequestRoles(requestRoles []user.Role) *Builder {
+	b.p.requestRoles = requestRoles
```
Comment (rot1024) [inline server/pkg/project/builder.go:91]:
```suggestion
	b.p.requestRoles = slices.Clone(requestRoles)
```

I have made this same point sometimes. Do you understand why this is necessary?

File: server/pkg/project/project.go [hunk 2]
```diff
@@ -93,6 +99,10 @@ func (p *Project) UpdateDescription(description string) {
 	p.description = description
 }
 
+func (p *Project) SetRequestRoles(sr []user.Role) {
+	p.requestRoles = sr
```
Comment (rot1024) [inline server/pkg/project/project.go:103]:
```suggestion
	p.requestRoles = slices.Clone(sr)
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: server/pkg/project/project.go (hunk 2 above)
Comment (nourbalaha) [reply server/pkg/project/project.go:103]: Done!

File: server/pkg/project/builder.go (hunk 1 above)
Comment (nourbalaha) [reply server/pkg/project/builder.go:91]: Yes, I understand why this is necessary. I am sorry about forgetting to add `slices.Clone()`.
Done!

**Review by rot1024 — APPROVED** [review]

**Pull request merged**